Add and reuse example for FileIO stream operator.

tz70s opened the pull request:
Add examples for both `fromPath` and `toPath` stream operators.

With an additional minor change in broken scala doc links.

Ref to #25468 

Due to `fromFile` and `toFile` operators are deprecated, maybe the pending operators can then be removed?

https://github.com/akka/akka/blob/2a5f254f16ccb6f39f60982bc7b11d1aaf137fe3/project/StreamOperatorsIndexGenerator.scala#L109-L112

tz70s committed:
Add and reuse example for FileIO stream operator.

Add examples for both fromPath and toPath stream operators.

With an additional minor change in broken scala doc links.

johanandren reviewed (commented):
LGTM

Re your question I think it could be good to have pages for the deprecated file factories, but they could maybe just contain a reference to the `Path` ones.

raboof reviewed (approved):
:+1: thanks!

tz70s committed:
Add fromFile and toFile pages as well.

But mark them as deprecated methods and point to fromPath and toPath operators.

tz70s commented:
I've added `toFile` and `fromFile` pages as well, with deprecation warning and ref to `toPath` and `fromPath` operators 👍

johanandren reviewed (approved):
LGTM, great that you did the file ones as well, thanks!

tz70s committed:
Small typo fix - emit to emits.

raboof commented:
OK TO TEST

raboof commented:
Thanks!
